src/components/forms/auto-complete/RuiAutoComplete.vue

```diff
@@ -0,0 +1,757 @@
+<script lang="ts" setup>
+import { logicAnd, logicOr } from '@vueuse/math';
+import RuiButton from '@/components/buttons/button/Button.vue';
+import RuiIcon from '@/components/icons/Icon.vue';
+import RuiChip from '@/components/chips/Chip.vue';
+import RuiMenu, { type MenuProps } from '@/components/overlays/menu/Menu.vue';
+import type { Ref } from 'vue';
+
+export type T = any;
+
+export type K = keyof T;
+
+export type ModelValue<T> = T | null | T[];
+
+export interface Props {
+  options: T[];
+  keyAttr?: K;
+  textAttr?: K;
+  value?: ModelValue<T>;
+  disabled?: boolean;
+  readOnly?: boolean;
+  dense?: boolean;
+  clearable?: boolean;
+  label?: string;
+  menuOptions?: MenuProps;
+  labelClass?: string;
+  menuClass?: string;
+  itemClass?: string;
+  prependWidth?: number; // in rem
+  appendWidth?: number; // in rem
+  itemHeight?: number; // in px
+  variant?: 'default' | 'filled' | 'outlined';
+  hint?: string;
+  errorMessages?: string | string[];
+  successMessages?: string | string[];
+  returnPrimitive?: boolean;
```

Comment:
Like before we should not expose that to the consumer, just infer it from the key being there or not